From 98913b4c1e12f2452988831a4b13bdbee1cf526c Mon Sep 17 00:00:00 2001
From: "Fomenko, Evarist M" <evarist.m.fomenko@intel.com>
Date: Tue, 15 May 2018 16:32:59 +0000
Subject: [PATCH] src: jit: do not ignore the result of fwrite...

... at least from compiler point of view.
On some systems with old glibc (<= 2.15) the fwrite function has
the warn_unused_result attribute. That makes MKL-DNN build fail.
Starting with glibc 2.16 the attribute was removed [1].

Alas, no regression test can be implemented here.

This closes #236.

[1]
The bug: https://sourceware.org/bugzilla/show_bug.cgi?id=11959
Release notes to 2.16:
https://sourceware.org/ml/libc-alpha/2012-06/msg00807.html
---
 src/cpu/jit_generator.hpp | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/cpu/jit_generator.hpp b/src/cpu/jit_generator.hpp
index 49204e0..4c730e1 100644
--- a/src/cpu/jit_generator.hpp
+++ b/src/cpu/jit_generator.hpp
@@ -719,7 +719,8 @@ public:
             FILE *fp = mkldnn_fopen(fname, "w+");
             // Failure to dump code is not fatal
             if (fp) {
-                fwrite(code, getSize(), 1, fp);
+                size_t unused = fwrite(code, getSize(), 1, fp);
+                UNUSED(unused);
                 fclose(fp);
             }
         }
-- 
2.17.0

